runtime/src/iree/hal/string_util.c

```diff
@@ -441,6 +465,38 @@ static iree_status_t iree_hal_parse_element_unsafe(
       *(uint8_t*)out_data = (uint8_t)iree_math_f32_to_f8e4m3fnuz(temp_float);
       return iree_ok_status();
     }
+    case IREE_HAL_ELEMENT_TYPE_FLOAT_8_E8M0_FNU: {
+      float temp_float = 0;
+      if (!iree_string_view_atof(data_str, &temp_float)) {
+        return iree_status_from_code(IREE_STATUS_INVALID_ARGUMENT);
+      }
+      *(uint8_t*)out_data = (uint8_t)iree_math_f32_to_f8e8m0fnu(temp_float);
+      return iree_ok_status();
+    }
+    case IREE_HAL_ELEMENT_TYPE_FLOAT_6_E3M2_FN: {
+      float temp_float = 0;
+      if (!iree_string_view_atof(data_str, &temp_float)) {
+        return iree_status_from_code(IREE_STATUS_INVALID_ARGUMENT);
+      }
+      *(uint8_t*)out_data = (uint8_t)iree_math_f32_to_f6e3m2fn(temp_float);
```

Comment:
I can't tell if this'll add in padding bits that aren't supposed to be there.

That is, f6 and f4 are densely packed the way that i6 and i4 would be, and ... do we need to handle that here?

Reply:
another very good point and catch - in general we don't handle sub-byte types at runtime and rely on users to handle that by treating them as binary blobs. slipped my mind. pretty sure the world will explode trying to print/parse these subbyte types. the test functions and such too are expecting byte-aligned values and won't work.

so, I don't think we can have any of these subbyte types in the HAL today (just the f8) - and maybe ever - as they are not well-defined and our tooling (what having a type known to the HAL is for) will not accept them.

darn - well, maybe back to the original PR then just adding the f8 type - I didn't expect the subbyte types to appear (just other f8 types) and missed that.